wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -656,6 +667,41 @@ public void deleteProject(@ApiParam("The workspace id")
         }
     }
 
+    @GET
+    @Path("/{id}/check")
```

Comment:
This path is not very REST compliant. Consider **/{id}/ws-agent/health**, **/{id}/ws-agent/availability**. @gazarenkov WDYT?